Automattic/wp-calypso
# Magic Login: Pass blog id to email endpoint

Branched from and depends on **this being merged first**: https://github.com/Automattic/wp-calypso/pull/83928
Also needs to be tested alongside this phab diff to take effect: D128131-code

## Proposed Changes

* Adds a blog_id param to the sendEmailLogin method, which passes it onward to the email endpoint used to send emails for magic email login flows.
* We recently deployed backend code (D128131-code) that will use the blog Id to customize magic login emails by using the customer site icon and site name in the email (rather than using the WordPress.com name and icon).
* See figma designs here: 5p3iZ99zPYfl6ronoTYEXO-fi-2_230

**Before**
<img width="1144" alt="email-before" src="https://github.com/Automattic/wp-calypso/assets/21228350/1632878c-ca4f-479c-9bfe-a7ea4bd16603">

**After**
<img width="1132" alt="email-after" src="https://github.com/Automattic/wp-calypso/assets/21228350/e35a280f-49b9-435e-9c89-8e5ac4d76541">

## Testing Instructions

**Setup**
1) You'll need a WordPress.com simple site that has a site logo/icon set, and has a blog post with a paywall block. 
2) In an incognito or non-logged-in browser, go to the front end post with the paywall, and click the 'already a subscriber' link. That will take you to the magic email screen. 
3) On the magic email screen replace https://wordpress.com with http://calypso.localhost:3000/ to load this branch. 
4) Enter an email, click enter, then go check your email inbox to confirm that the email has been customized with the users site icon and site title like the screenshot above. 

**Testing**
* In a non-logged in browser, open a blog post with a paywall block on your simple site
* Click the 'Already a subscriber' link, which will take you to the magic email screen
* On the magic email screen replace https://wordpress.com with http://calypso.localhost:3000/ to load this branch. 
* Test 1: On the magic email screen enter an email not associated with any WordPress account
   - Check the email in your inbox, and confirm that it shows the site icon and site title like screenshot above. 
* Test 2: On the magic email screen enter an email not associated with any WordPress account
   - Check the email in your inbox, and confirm that it shows the site icon and site title like the screenshot above.
   
Note: Another diff ( 

## Pre-merge Checklist
